File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03630/p03630_num9_parsed.py
def max_rectangle_area(H, W, grid):
    left = [0] * W
    right = [W] * W
    height = [0] * W
    max_area = 0

    for i in range(H):
        cur_left = 0
        cur_right = W
        for j in range(W):
            if grid[i][j] == '#':
                height[j] = 0
                left[j] = 0
                cur_left = j + 1
            else:
                height[j] += 1
                left[j] = max(left[j], cur_left)

        for j in range(W - 1, -1, -1):
            if grid[i][j] == '#':
                right[j] = W
                cur_right = j
            else:
                right[j] = min(right[j], cur_right)

        for j in range(W):
            max_area = max(max_area, height[j] * (right[j] - left[j]))

    return max_area


H, W = map(int, input().split())
grid = [input() for _ in range(H)]
print(max_rectangle_area(H, W, grid))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03630/p03630_num9_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int H, W;
    cin >> H >> W;

    vector<string> grid(H);
    for (int i = 0; i < H; i++) {
        cin >> grid[i];
    }

    vector<vector<int>> prefix_black(H + 1, vector<int>(W + 1, 0));
    for (int i = 0; i < H; i++) {
        for (int j = 0; j < W; j++) {
            prefix_black[i + 1][j + 1] = prefix_black[i + 1][j] + prefix_black[i][j + 1] - prefix_black[i][j];
            if (grid[i][j] == '#') {
                prefix_black[i + 1][j + 1]++;
            }
        }
    }

    int max_area = 0;
    for (int top = 0; top < H; top++) {
        for (int left = 0; left < W; left++) {
            for (int bottom = top + 1; bottom < H + 1; bottom++) {
                for (int right = left + 1; right < W + 1; right++) {
                    int total_black = prefix_black[bottom][right] - prefix_black[top][right] - prefix_black[bottom][left] + prefix_black[top][left];
                    if (total_black == (bottom - top) * (right - left)) {
                        max_area = max(max_area, total_black);
                    }
                }
            }
        }
    }

    cout << max_area << endl;

    return 0;
}